plugins/org.jkiss.dbeaver.data.transfer.ui/src/org/jkiss/dbeaver/tools/transfer/ui/pages/database/ConfigureMetadataStructureDialog.java

```diff
@@ -0,0 +1,420 @@
+                        for (Map.Entry<String, Object> entry : rawChangedPropertiesMap.entrySet()) {
+                            DBPPropertyDescriptor property = propertySource.getProperty(entry.getKey());
+                            if (property != null) {
+                                propertySource.addChangedProperties(property, entry.getValue());
+                            }
+                        }
+                        // Update table properties
+                        DatabaseTransferUtils.implementPropertyChanges(
+                            null,
+                            propertySource.getChangedPropertiesValues(),
+                            null,
+                            null,
+                            (DBSEntity) tableObject);
+                    }
+
+                    PropertyTreeViewer propertyViewer = new PropertyTreeViewer(configTabs, SWT.BORDER);
+                    propertyViewer.getControl().setLayoutData(GridDataFactory.fillDefaults().hint(400, SWT.DEFAULT).create());
+                    propertyViewer.loadProperties(propertySource);
+
+                    tablePropertiesTab.setControl(propertyViewer.getControl());
+                } else {
+                    createCompositeWithMessage(gd, tablePropertiesTab, DTUIMessages.page_configure_table_properties_no_properties);
+                }
+            } else {
+                createCompositeWithMessage(gd, tablePropertiesTab, DTUIMessages.page_configure_table_properties_info_text);
+            }
+        }
+
+        TabItem showDDLTab = new TabItem(configTabs, SWT.NONE);
+        showDDLTab.setText(DTMessages.data_transfer_wizard_page_ddl_name);
+        showDDL(showDDLTab);
+
+        configTabs.setSelection(0);
+        configTabs.addSelectionListener(new SelectionAdapter() {
+            @Override
+            public void widgetSelected(SelectionEvent e) {
+                if (ddlTabNeedRefresh) {
+                    final int selectionIndex = configTabs.getSelectionIndex();
+                    final Control[] tabList = configTabs.getTabList();
+                    if (tabList.length > 0 && selectionIndex == tabList.length - 1) {
+                        // Refresh DDL tab - it is the last
+                        final DBSObjectContainer container = settings.getContainer();
+                        if (container == null) {
+                            return;
+                        }
+                        DBPDataSource dataSource = container.getDataSource();
+                        setNewTextToDDLTab(container, dataSource);
+                    }
+                }
+            }
+        });
+
+        return composite;
+    }
+
+    private void createCompositeWithMessage(GridData gd, TabItem tablePropertiesTab, String message) {
+        Composite compositeEmpty = new Composite(configTabs, SWT.NONE);
+        compositeEmpty.setLayout(new GridLayout(1, false));
+        compositeEmpty.setLayoutData(gd);
```

Comment:
I'm unsure about the point of having two nested placeholder composites. Can we avoid this?

Reply:
Nope